pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/exchange/BlockExchange.java

```diff
@@ -68,25 +67,29 @@ protected BlockExchange(List<SendingMailbox> sendingMailboxes, BlockSplitter spl
     _splitter = splitter;
   }
 
-  public void send(TransferableBlock block)
+  /**
+   * API to send a block to the destination mailboxes.
+   * @param block the block to be transferred
+   * @return true if any of the upstream mailboxes requested EOS (e.g. early termination)
```

Comment:
This comment is incorrect. It returns true only if all receiving mailboxes are early terminated.

Side comment: Do you realize this upstream is the opposite of other upstream changes in the PR (in this place it refers to the next operator in the data flow, but in other places it refers to the previous operator in the data flow). Thus suggest not using upstream because it can be confusing

Reply:
yeah it is confusing, will create a separate PR for naming fix